### DISCUSSION TITLE
CLICK TO DIAL

### DISCUSSION THREAD

**Discussion by bright123-strive** [InnovateAsterisk/Browser-Phone#487] [Q&A]
Hi am creating a click to dial using php and the browser phone. but my problem is my script is fully executed but the call does not reach the browser phone. but the extension is registered correctly and i can call someone by typing the number or another extension. here is my php code  
<?php
#ip address that asterisk is on.
$strHost = "127.0.0.1";
$strUser = "sipline";#specify the asterisk manager username you want to login with
$strSecret = "sipline12";#specify the password for the above user
#specify the channel (extension) you want to receive the call requests with
#e.g. SIP/XXX, IAX2/XXXX, ZAP/XXXX, etc
# $strChannel = "SIP/100";
$strChannel = $_REQUEST['exten'];
$strContext = "from-internal";
#specify the amount of time you want to try calling the specified channel before hangin up
$strWaitTime = "30";
#specify the priority you wish to place on making this call
$strPriority = "1";
#specify the maximum amount of retries
$strMaxRetry = "2";
$number=strtolower($_REQUEST['number']);
$pos=strpos ($number,"local");
if ($number == null) :
exit() ;
endif ;
if ($pos===false) :
$errno=0 ;
$errstr=0 ;
$strCallerId = "Web Call $number";
$oSocket = fsockopen("localhost", 5038, $errno, $errstr, 20);
if (!$oSocket) {
echo "$errstr ($errno)<br>\n";
} else {
fputs($oSocket, "Action: login\r\n");
fputs($oSocket, "Events: off\r\n");
fputs($oSocket, "Username: $strUser\r\n");
fputs($oSocket, "Secret: $strSecret\r\n\r\n");
fputs($oSocket, "Action: originate\r\n");
fputs($oSocket, "Channel: $strChannel\r\n");
fputs($oSocket, "WaitTime: $strWaitTime\r\n");
fputs($oSocket, "CallerId: $strCallerId\r\n");
fputs($oSocket, "Exten: $number\r\n");
fputs($oSocket, "Context: $strContext\r\n");
fputs($oSocket, "Priority: $strPriority\r\n\r\n");
fputs($oSocket, "Action: Logoff\r\n\r\n");
sleep(2);
fclose($oSocket);
}
echo "Extension $strChannel should be calling $number." ;
else :
exit() ;
endif ;
?>
![Screenshot (43)](https://github.com/InnovateAsterisk/Browser-Phone/assets/82960039/750989e7-c6d2-49b5-bfc2-aca11dd86c23)

![Screenshot (44)](https://github.com/InnovateAsterisk/Browser-Phone/assets/82960039/c06cabaf-ee0e-4d63-bf83-fea12474c716)

**Comment by InnovateAsterisk** [comment]
Looking great!! However, from the screenshots you provided it’s looks like you have embedded the phone into the global scope of your page. This is not the recommended way, as the phone doesn’t play well with existing pages. (As you also can see in the CSS issues)

The recommended solution is to open the phone as an IFRAME.

https://github.com/InnovateAsterisk/Browser-Phone/blob/master/Phone/popup.html

the above is a basic solution that opens a click-to-dial page. (also available in the same folder)

You have some choices on how you want it to behave. These pages are very simple and don’t cater for all the scenarios, so you will have to do a bit of your own work to get it to work. For example, do you want the phone to remain registered when the phone is hidden? Or must it close and de-register? Also, it if opens on making a call, must it close after? All these kids of options are for your own design.

> my problem is my script is fully executed but the call does not reach the browser phone. but the extension is registered correctly and i can call someone by typing the number or another extension.

If the browser phone shows as registered, but asterisk sees it as unavailable, then Asterisk is saving the contact information as-is, probably something like 192.169.0.1 or some impossible ip. This means that after the registration, Asterisk sends an options packet, but it will not reach the client. So it marks the endpoint as unavailable. To solve this make sure asterisk is set to re-write contacts.

if your contact is fine, and asterisk is able to qualify your endpoint, but calls fail on the client. (You may get a missed call notification in the browser phone), then this is because a script failed in the browser phone. You will need to inspect the developer console to understand the issue.

**Reply by bright123-strive** [reply]
Thanks a lot sir this was much helpful I was really getting a missed call.
